packages/api-server/src/methods/modules/poly.ts

```diff
@@ -87,15 +94,50 @@ export class Poly {
 
   async executeRawL2Transaction(args: any[]) {
     try {
-      const data = args[0];
-      const txWithAddressMapping: RawL2TransactionWithAddressMapping =
-        deserializeRawL2TransactionWithAddressMapping(data);
-      const rawL2Tx = txWithAddressMapping.raw_tx;
+      const serializeRawL2Tx = args[0];
+
+      const executeCallResult = async () => {
+        const txWithAddressMapping: RawL2TransactionWithAddressMapping =
+          deserializeRawL2TransactionWithAddressMapping(serializeRawL2Tx);
+        const rawL2Tx = txWithAddressMapping.raw_tx;
+        const jsonResult = await this.rpc.executeRawL2Transaction(rawL2Tx);
+        // if result is fine, then tx is legal, we can start thinking to store the address mapping
+        await saveAddressMapping(this.query, this.rpc, txWithAddressMapping);
+        const stringResult = JSON.stringify(jsonResult);
+        return stringResult;
+      };
 
-      const result = await this.rpc.executeRawL2Transaction(rawL2Tx);
-      // if result is fine, then tx is legal, we can start thinking to store the address mapping
-      await saveAddressMapping(this.query, this.rpc, txWithAddressMapping);
-      return result;
+      // using cache
+      if (envConfig.enableCachePolyExecuteRawL2Tx === "true") {
+        // calculate data cache key
+        const tipBlockHash = await this.rpc.getTipBlockHash();
+        const memPollStateRoot = await this.rpc.getMemPoolStateRoot();
```

Comment:
maybe we could use Promise.all([tipBlockHash, memPollStateRoot]) ?

Reply:
> maybe we could use Promise.all([tipBlockHash, memPollStateRoot]) ?

yea, that will be faster

Reply:
updated